include/substrait-mlir/Dialect/Substrait/IR/SubstraitTypes.td

```diff
@@ -25,6 +25,24 @@ class Substrait_Attr<string name, string typeMnemonic, list<Trait> traits = []>
   let mnemonic = typeMnemonic;
 }
 
+def Substrait_UUIDType : Substrait_Type<"UUID", "uuid"> {
+  let summary = "Substrait uuid type";
+  let description = [{
+    This type represents a substrait uuid (universally-unique identifier) type.
+    This is a 16-byte binary value.
+  }];
+}
+
+def Substrait_UUIDAttr : Substrait_Attr<"UUID", "binary"> {
+  let summary = "Substrait uuid type";
+  let description = [{
+    This type represents a substrait uuid (universally-unique identifier) attribute type.
+    This is a 16-byte binary value.
+  }];
+  let parameters = (ins ArrayRefParameter<"uint8_t", "">:$value);
+  let assemblyFormat = [{ `<` $value `>` }];
```

Comment:
It'd be nice to (eventually) print the UUIDs in the typical format (see [Wikipedia](https://en.wikipedia.org/wiki/Universally_unique_identifier)). Doesn't need to be now; can just be a TODO.

If you do want to go for it, the `APInt` class from LLVM probably has a few interesting features:
* Construct from array: [constructor](https://github.com/llvm/llvm-project/blob/19c516c8d5/llvm/include/llvm/ADT/APInt.h#L148)
* Extract bits at certain positions: [`extractBits`](https://github.com/llvm/llvm-project/blob/19c516c8d/llvm/include/llvm/ADT/APInt.h#L1460).
* Convert to hex string: [`toString`](https://github.com/llvm/llvm-project/blob/19c516c8d5716c/llvm/include/llvm/ADT/APInt.h#L1663) with `radix` set to 16.